applications/wordpress-microservice/templates/mariadb-deployment.yaml

```diff
@@ -12,11 +12,39 @@ spec:
   template:
     metadata:
       labels:
+       #{{- include "wordpress-microservice.selectorLabels" . | nindent 8 }}
         app: mariadb
+      annotations:
+        checksum/secret: {{ include (print $.Template.BasePath "/secret.yaml") . | sha256sum }}
     spec:
+      {{- if .Values.mariadb.securityContext.enabled }}
+      securityContext:
+        runAsUser: {{ .Values.mariadb.securityContext.runAsUser }}
+        fsGroup: {{ .Values.mariadb.securityContext.fsGroup }}
+      {{- end }}
```

Comment:
A common pattern is to distinguish between security context at the pod level vs. the container level. 

Also, i'm not sure if you're allowing every possible option to be set which might cause you more work in the future. 

See how this is handled in Mlflow to get a better idea for how I've commonly seen this implemented - https://github.com/replicatedhq/platform-examples/blob/main/applications/mlflow/charts/mlflow/templates/deployment.yaml#L59-L62

You'll notice there is a `podSecurityContext` and `containerSecurityContext` value